java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java

```diff
@@ -554,6 +554,7 @@ public void nextVector(ColumnVector previousVector,
         longColVector = (LongColumnVector) previousVector;
       } else {
         decimalColVector.ensureSize(batchSize, false);
+        decimalColVector.reset();
```

Comment:
Should you revert the changes to `ConvertTreeReaderFactory.java` ?
There is now code to check and set `isRepeating`.
After reverting, decimal will be handled in the same way as the other types.